jobs/worker/spec

```diff
@@ -258,6 +258,10 @@ properties:
     description: |
       Time to wait for requests to Containerd to complete. 0 means no timeout.
 
+  containerd.init_bin_path:
+    env: CONCOURSE_CONTAINERD_INIT_BIN
+    description: |
+      Path to a dumb init executable
```

Comment:
Hmm, I wonder if we should even surface this flag at all. For deployments like BOSH where we have full control over where stuff goes, is it even possible for the init binary to be deposited somewhere else?

@andy-paine what are your thoughts on this being configurable?

Reply:
I can't really see a world in which I would ever want to override it so just setting it to the right value in the template seems all good to me. I mostly wanted it configurable through the binaries rather than the releases so the difference in paths between Docker (k8s) and BOSH could be reflected.

Reply:
tldr: for consistency I would keep this around. I've added a comment noting what the value defaults to when containerd is the runtime.

That makes sense, and the thought did occur to me.

For the most part we try to ensure whatever is exposed in the binary options is exposed in the release and chart as well. The intention being operators always have the right knobs for all the new features implemented. There are jobs in CI that check this.

Making exceptions using the `ignored-in-concourse` and `ignored-in-distribution` files has gotten us into a lot of trouble as the logic for exceptions become tricky for people to standardize around. I've recently been trying to [streamline](https://github.com/concourse/concourse/issues/5985#issuecomment-692202317) this and have the approach of using as few variables in those files as possible as well as have principles around what can go in.

Reply:
I guess my main concern is that we're exposing something thing that shouldn't be touched. Changing it to any other value is invalid and that's pretty counter intuitive to me: if there's only a single option for the configuration, why is it configurable in the first place?

I wonder if we can do something like putting it in the spec file, but have the template ignore any values provided and just use a hard coded location. But that just feels like a very elaborate way of emulating `ignored-in-concourse`

Reply:
Although, now that I think about it, this *technically* allows the cluster operator to create their own init binary with custom logic and have that be injected into every single container created by Concourse 🤔

Reply:
@chenbh I see four reasons for keeping this configurable.

**Consistency Without Risk** - I'm trying to understand your concern about exposure. Do you think it would lead to faulty deployments? Practically speaking, I do not think any operator would set this flag to a random value without having a very specific purpose like the use case you mentioned. I feel the benefit of consistency outweighs that highly hypothetical risk. Let me know if I'm missing something here. 

**Potential Use Cases** - There is the use case you mentioned. Here's another use case. While BOSH might not be seeing a lot of new features, theoretically the behaviour could change where the concourse bin files get moved to a new location like `var/vcap/lib` rather than `/var/vcap/packages`. In such a case the operator would need to set init-bin to the new path. Ultimately, we can't predict every use case, and barring security concerns, IMO something that is already configurable should be left open for users to tinker with.

**Potentially Configurable Anyway** - I can't tell if BOSH, like Kubernetes, allows the operator to pass any environment variables they need to directly. If it did then the init-bin would be configurable regardless of us keeping it out of the spec.

**Easily Accessible Documentation** - I think it's useful to be transparent in the spec about what's available in the binary. One of the reasons we document all our configuration in the spec is so that operators don't have to download a tarball of the binary, unzip it, put it on Linux if they aren't on it, and run the binary to see the full list of options. This is something that has actually happened. Chances are that they may spot `CONCOURSE_CONTAINERD_INIT_BIN` in the templating script, or in an error message when things go wrong, and can't easily look up from the spec what it means.

Reply:
Ok, the consistency part sold me